internal/date/date.go

```diff
@@ -22,7 +22,7 @@ type Date struct {
 	month string
 	day   string
 
-	t   time.Time
+	T   time.Time
```

Comment:
Exporting this so when we check object hashes at the start of a `Put` there is something to compare against.

Reply:
I'd prefer to do https://pkg.go.dev/github.com/mitchellh/hashstructure#Hashable, which could be as simple as 

```
func (d Date) Hash() (uint64, error) { return uint64(d.t.Unix()), nil }
```